### PR TITLE
[1LP][RFR]added ip and host_credentials check in cleanup scripts.

### PR DESCRIPTION
**Pull request by amavinag**
Purpose or Intent
=================
Credentials check is necessary when host_credentials are added to cfme_data, otherwise default to host_default credentials. Similarly, IP check is necessary for the providers in cleanup script, if IP is not reachable then there is no necessary to try invoking that provider.

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 0e509cb7124e571e8852dd596561fe2151f53be0:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

File: scripts/cleanup_old_vms.py [hunk 1]
```diff
@@ -147,6 +149,9 @@ def cleanup_vms(texts, max_hours=24, providers=None, prompt=True):
     matchers = [re.compile(text) for text in texts]
 
     for provider_key in providers:
+        ipaddress = cfme_data['management_systems'][provider_key].get('ipaddress', None)
```
Comment (mshriver) [inline scripts/cleanup_old_vms.py]: :+1:

File: scripts/sync_template_tracker.py [hunk 2]
```diff
@@ -22,6 +22,9 @@ def main(trackerbot_url, mark_usable=None):
     unresponsive_providers = set()
     # Queue up list_template calls
     for provider_key in all_providers:
+        ipaddress = cfme_data['management_systems'][provider_key].get('ipaddress', None)
```
Comment (psav) [inline scripts/sync_template_tracker.py]: Won't this prevent us from tracking templates on thins like azure and ec2? as they don't have IP addresses? Though, I don't think we track those any way do we?

Comment (psav) [inline scripts/sync_template_tracker.py]: However the comment may be related to the above usage

**Pull request merged**